src/pages/studyView/StudyViewUtils.spec.ts

```diff
@@ -3442,4 +3443,108 @@ describe('StudyViewUtils', () => {
             ]);
         });
     });
+
```

Comment:
do we test the uniquness anywhere?

Reply:
Yes. The third test has study `S01` and all of its samples in both groups. Without proper uniqueness enforcement, I would expect the three S01 samples to appear twice.